src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileValidation.java

```diff
@@ -302,6 +322,258 @@ public ValidationOutputData validateMandatoryFieldsAndData(PerformanceProfile pe
 
         return validationOutputData;
     }
+
+    /**
+     * Validates the data present in the metric profile object before adding it to the map
+     * @param metricProfile
+     * @return
+     */
+    private ValidationOutputData validateMetricProfileData(PerformanceProfile metricProfile) {
```

Comment:
can you merge some of the very common functionalities likes
 // Check if k8s type is supported
 SLO validations etc...  with
 validatePerformanceProfileData

Reply:
validation of common fields like `k8s_type` & `slo` is moved to `validateCommonProfileFields()`